pkg/pushsync/pushsync.go

```diff
@@ -361,33 +336,25 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin bo
 		}
 	}
 
-	retry()
-
-	// nextPeer attempts to lookup the next peer to push the chunk to, if there are overdrafted peers the boolean would signal a re-attempt
-	nextPeer := func() (peer swarm.Address, err error) {
-
-		fullSkipList := append(skip.ChunkPeers(ch.Address()), ps.skipList.ChunkPeers(ch.Address())...)
-
-		peer, err = ps.topologyDriver.ClosestPeer(ch.Address(), includeSelf, topology.Filter{Reachable: true}, fullSkipList...)
-		if err != nil {
-			if errors.Is(err, topology.ErrWantSelf) {
-				if !ps.warmedUp() {
-					return swarm.ZeroAddress, ErrWarmup
-				}
+	nextPeer := func() (swarm.Address, error) {
 
-				if !ps.radiusChecker.IsWithinStorageRadius(ch.Address()) {
-					return swarm.ZeroAddress, ErrOutOfDepthStoring
-				}
-
-				ps.pushToNeighbourhood(ctx, fullSkipList, ch, origin, originAddr)
-				return swarm.ZeroAddress, err
-			}
+		// we are a full node and in the neighborhood of the chunk
+		if ps.fullNode && ps.radiusChecker.IsWithinStorageRadius(ch.Address()) {
```

Comment:
Before checking the radius and returning ErrWantSelf, we should also ensure we are warmed up?

I guess now the radius starts at higher number and then comes down so this is okay?